### ISSUE TITLE
Create project from a private repo

### ISSUE THREAD

**Issue by gaow** [bioworkflows/bioworkflows-help#22]
I tried to create a repository from a folder in one of my private github repo (this folder: https://github.com/cumc/bioworkflows/tree/master/GWAS). It says:

![image](https://user-images.githubusercontent.com/917985/85963318-05f4b900-b983-11ea-8000-276476ae274a.png)

**Comment by BoPeng** [comment]
Please go to settings, disconnect GitHub and reconnect. This will grant more permissions to the user to read private repo.

**Comment by gaow** [comment]
When I go to settings I only see a button "Connect to github account" and when I click on it I get redirected to bioworkflows.com/gaow page. I dont mind deleting my account and start again. But i dont see an option to delete my account either.

**Comment by BoPeng** [comment]
This is when I am already connected.

![image](https://user-images.githubusercontent.com/9889312/85966169-8f58bb00-b984-11ea-845f-66388b9c10dc.png)

In your case, you are not connected. Let me disconnect and see if the redirect works.

**Comment by BoPeng** [comment]
Yes, the connect button does not work. What if you log out and log in with github?

**Comment by BoPeng** [comment]
Try again.

**Comment by gaow** [comment]
Okay signing in with github works although the experience was not smooth, and i ended up with a username not my github but my gmail prefix (`wangow`).

Let me try log out and log back in without github -- seems you fixed it?

**Comment by gaow** [comment]
> Try again.

Nope, same problem. I cannot connect to github from my `gaow` account.

**Comment by BoPeng** [comment]
But at least you have linked your bioworkflows username to the github one, right? Third-party login is not easy. I barely got it working and have not streamlined the process or interface.

**Comment by gaow** [comment]
> you have linked your bioworkflows username to the github one, right? 

Now it seems i have two accounts. Once usernames is `wangow` and is named so because i logged in with github and was assigned username based the gmail account associated with my github. The other account, `gaow`, is what I registered directly at bioworkflow. It does not know anything about my github account.

**Comment by BoPeng** [comment]
> Now it seems i have two accounts. 

That is not right. You are supposed to link the two. Say, inside bioworkflows `gaow` link to your github account... Let me remove your second account first.

**Comment by BoPeng** [comment]
Now, log in with `gaow` and try to connect from `settings`?

**Comment by gaow** [comment]
> nside bioworkflows gaow link to your github account

The problem is I still cannot do that. Same problem -- when i request to connect i get redirected to homepage.

> Let me remove your second account first.

Good. please let me know when the first account works with linking. I don't see the same info as your screen shot above, though.

**Comment by BoPeng** [comment]
I see 

![image](https://user-images.githubusercontent.com/9889312/85967441-14919f00-b988-11ea-80c0-ce385ce2b9fa.png)

after I disconnect from github and click connect button. Perhaps you need to refresh history or change browser?

**Comment by gaow** [comment]
No luck with google chrome private mode, nor with firefox.

It's just `settings` then `profile -> connect to github`? Perhaps you can delete that account too and let me know. I'll register again and see.

**Comment by gaow** [comment]
Here is screenshot on my firefox. I dont use firefox for other than tests like this.

![image](https://user-images.githubusercontent.com/917985/85967892-815d6700-b991-11ea-99f7-4b77db57b695.png)

**Comment by BoPeng** [comment]
I should not have removed your another account, because over there you can see if you can connect and disconnect to github.

I will spare you with the tests and create a fake github account to test more. I will let you know what I find out.

**Comment by BoPeng** [comment]
Let me know if the connect button works now.

**Comment by gaow** [comment]
@BoPeng it still does not work for me! So I asked my colleague to try. I just got the feedback -- same issue, it redirects to the user's workflow page after clicking the `Connect to your GitHub Account` button.

**Comment by BoPeng** [comment]
Thanks. Let me try more. It was quite mysterious to me because it suddenly worked (after a url change) but I then worked even if I change back the url etc.

**Comment by BoPeng** [comment]
What happens if you type in the following two links directly after login?

https://bioworkflows.com/~accounts/social/connections/
https://bioworkflows.com/~accounts/github/login/?process=connect

**Comment by gaow** [comment]
The first link gives:

![image](https://user-images.githubusercontent.com/917985/86024374-ea2cf980-b9fa-11ea-89d5-a94ef3ff42c0.png)


the 2nd link gives

![image](https://user-images.githubusercontent.com/917985/86024438-fd3fc980-b9fa-11ea-88de-fb0895c8b434.png)

**Comment by gaow** [comment]
And after clicking on these I was able to import pipelines from private repo

**Comment by BoPeng** [comment]
So maybe something went wrong with the generation of the URL (the server needs to look up what third-party app is available to calculate the URL). Let me rework this part (providing the direct link the github instead of this page) before you ask your friend to try again. The system has memory and we need fresh account to test.
